docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,488 @@
+this library into OE SDK as part of enclave-side plugin.
+
+```C
+oe_result_t oe_verify_qve_report_and_identity(
+        const uint8_t *p_quote,
+        uint32_t quote_size,
+        const sgx_ql_qe_report_info_t *p_qve_report_info,
+        time_t expiration_check_date,
+        uint32_t collateral_expiration_status,
+        sgx_ql_qv_result_t quote_verification_result,
+        const uint8_t *p_supplemental_data,
+        uint32_t supplemental_data_size,
+        sgx_isv_svn_t qve_isvsvn_threshold)
+```
+All the data returned by the QvE are included in the QvE report data field for
+integrity protection.
+```C
+QvE report_data = SHA256([nonce || quote || expiration_check_date ||
+expiration_status || verification_result || supplemental_data] || 32 - 0x00)
+```
+
+The flow of the API function `oe_verify_qve_report_and_identity()` is as below:
+- Verify QvE report.
+- Verify report data (report data described as above).
+- Check QvE Identity by comparing QvE report fields against hardcode values.
+  - Check that Report.MRSIGNER equals to Hardcoded QvE MRSIGNER.
+  - Check that Misc Select, Attribute and ProdID equal to the hardcoded values,
+    Misc select and Attribute need to apply Mask before comparison.
+- Check that Report.ISVSVN >= Hardcoded ISV SVN.
+
+Options to provide QvE ISV SVN threshold:
+- Option 1: Hardcode QvE ISV SVN in enclave-side plugin. Everytime QvE
+            ISV SVN is updated, the plugin library implementation will need to
+            be updated to stay in sync.
+- Option 2: Update existing enclave-side plugin API to allow the verifier to input
+            QvE ISV SVN threshold value. But it requires plugin API change, also
+            it's NOT TEE agnostic.
+
+We propose option 1 to avoid plugin API change.
+
+### Update claims list to add SGX Quote Verification status and QVL/QvE returned supplemental data
+In SGX remote attestation, the verifier may want to provide a different quote
+verification policy than the one enforced by the sgx_qv_verify_quote() API.
+
+The proposal is to extend claims definition with new SGX related
+claims.
+
+SGX quote verification API `sgx_qv_verify_quote()` returns 2 types of errors,
+fatal errors or non-fatal errors:
+- Fatal error: such as cert chain verification failed, cert revoked, API will
+  return an error code directly
+- Non-fatal error, such as cert chain out-of-date, API will return successful, but
+  there is an output parameter to indicate the specific error code. In this case,
+  the verifier can decide whehter the SGX quote is valid based on their own policy.
+
+We propose to introduce a new SGX-specific claim `OE_CLAIM_SGX_QUOTE_VERIFICATION_STATUS` to
+indicate QVL/QvE return status. The verifier can check the status to see if there
+is a non-fatal error during quote verification. This claim can have one of three values:
+
+  - OE_OK - Quote verification passed, and TCB level is up to date
+  - Non-Fatal error code - API returns OE_OK,
```

Comment:
I agree with you. 
But @yentsanglee pointed out that some customers may not care about the non-fatal (warning) errors, maybe return OE_OK is a good choice, then only the customer who want have a strict verification policy, he/she can check the additional status in SGX claims.

Reply:
In current OE SDK API design, if a function returns some error code other than `OE_OK`, does it always abort and not return other results in output parameters?

Reply:
Current OE SDK API design, OE_OK is the only success return value. Attestation APIs follow the same pattern. My only concern is backward compatibility. Otherwise, I generally agree with @CodeMonkeyLeet and @shnwc that the caller can make its own judgement by all the returned value.

Reply:
To recap a point in one of the SIG-attestation meetings: OE SDK currently makes specific policy decisions to ignore some verification issues on behalf of the caller. These decisions align with Azure expectations due to how it manages its SGX configuration, but should not be the default in a general SDK intended for multi-cloud/provider use like OE moving forwards. In that regard, a break with backwards compatibility is necessary here, and we should just manage it as we would other breaking changes to minimize the impact to the current users.